src/scripts/core/models/column.ts

```diff
@@ -1,40 +1,100 @@
 import { Container } from "pixi.js";
 import { Token } from "./token";
 
-
 export class Column extends Container{
 
-    private tokens: Token[];
+    public tokens: Token[] = [];
     private columnID: number;
     private columnSize: number;
-    // private viewWidth: number;
-    // private viewHeight: number;
-
+    private availHeight: number;
 
     constructor(columnID: number, columnSize: number, availWidth: number, availHeight: number) {
         super()
 
         this.columnID = columnID;
         this.columnSize = columnSize;
-        // this.viewWidth = viewWidth;
-        // this.viewHeight = viewHeight;
+        this.availHeight = availHeight;
 
-        this.tokens = [];
         for(var i = 0; i < this.columnSize; i++) {
-            const randomNumber = Math.round(Math.random() * (9 - 1) + 1);
-
-            //const newToken = new Token(ran, this.columnID, availWidth, columnSize,);
-            //parentID: number, locationIndex: number, size: number, skIndex: number, availWidth: number
-            const newToken = new Token(this.columnID, i, this.columnSize, randomNumber, availWidth)
-            newToken.y = (availHeight / (columnSize/i));
+            const newToken = new Token(this.columnID, i, this.columnSize, availWidth, availHeight)
+            newToken.y = (availHeight / (this.columnSize/i));
 
             this.tokens.push(newToken);
             this.addChild(newToken);
         }
     }
 
-    public getAllTokens(): Token[] {
-        return this.tokens;
+    public processMatches(): void {
+            //move tokens from original positions in to the furthest matched token position
+            this.columnTweens();
+
+            //rebuild and reset the column's tokens
+            this.rebuildTokens();
+
+    }
+
+    private columnTweens(): void {
+        const nonMatchCombo: Token [] = [];
+        for(let columnInspector: number = 0; columnInspector < this.tokens.length-1; columnInspector++) {
+            const target = this.tokens[columnInspector];
+            if(!target.matched) {
+                nonMatchCombo.unshift(target);
+                for(let combo = columnInspector+1; combo <= this.tokens.length-1; combo++) {
+                    if(this.tokens[combo].matched) { 
+                        for(let i = 0; i < nonMatchCombo.length; i++){
+                            nonMatchCombo[i].moveTo(combo-i);
+                        }
+                    }
+                    else {
+                        break;
+                    }
```

Comment:
This is purely personal preference so feel free to keep it as is if you prefer but for readability I think this would be slightly nicer as:

```ts
if (!this.tokens[combo].matched) { break; }

for (let i = 0; i < nonMatchCombo.length; i++) {
    nonMatchCombo[i].moveTo(combo-i);
}
```

I tend to group any escape clasues like that together before the code block that runs if it passes them. So it's clear that it must pass these escape clauses before the code block should happen. Again personal preference overall.